imperative.md

```diff
@@ -0,0 +1,23 @@
+# Should KCL be imperative?
+
+Currently KCl tries to keep all its variables immutable. Should we allow mutation?
+
+## Why immutable
+
+ - Fewer chances of bugs when variables cannot be updated
+ - Encourages users to use higher-level constructs like patterns, that can be optimized in both KCL and the engine much more easily than a for loop.
+ - Easy to map a specific geometric feature (e.g. a 2D polygon) to its definition in the IDE because a geometric feature is created once and then never changed, so it's fully defined at its source range. If geometry could be updated, we'd need the IDE to show where it was defined AND where the relevant change was made.
+ - KCL is about defining geometry, not about modeling a changing system. Mutation doesn't seem necessary.
+
```

Comment:
I'll add that it makes a lot of analyses (e.g., for the IDE), optimisations, and other reasoning about the code, much, much easier. In particular, partial execution (for sketch mode) and code mods will be much harder to get right in the presence of mutation